packages/components/table-v2/src/utils.ts

```diff
@@ -29,5 +29,6 @@ export const componentToSlot = <T>(
   ComponentLike: JSX.Element | ((props: T) => Component<T>) | undefined
 ) =>
   isVNode(ComponentLike)
-    ? (props: T) => h(ComponentLike, props)
+    ? // @ts-expect-error
```

Comment:
Refer to [this](https://github.com/element-plus/element-plus/pull/18331/commits/7922a03104cc2c77fb73dc7aaaf2429609c19eb4#diff-7d84f7ab3f84afb02294656109dfef8ac4d5c9d77c8f04731b28e047ce22a2abR28)